src/oauth.js

```diff
@@ -0,0 +1,60 @@
+const axios = require('axios');
+const Koa = require('koa');
+const Router = require('koa-router');
+const manifest = require('../manifest.json');
+
+const app = new Koa();
+const router = new Router();
+
+const TS_URL_BASE = 'https://api-smoketest.eu-west-1.test.ts.sv';
+const TS_URL_LOGIN = `${TS_URL_BASE}/tradeshift/auth/login`;
+const TS_URL_TOKEN = `${TS_URL_BASE}/tradeshift/auth/token`;
+
+const clientId = `${manifest.vendor_id}.${manifest.app_id}`;
+
+// Step 1: when a user opens the app, they hit the endpoint specified as `app.main` the manifest
+router.get('/', async (ctx) => {
+	const redirectUri = manifest.app.redirect_uri;
+
+	// Step 2: our app redirects to the oauth login endpoint on Tradeshift to kick off the oauth flow
+	const url = `${TS_URL_LOGIN}?response_type=code&client_id=${clientId}&redirect_uri=${redirectUri}&scope=openid offline`;
```

Comment:
Is this `&scope=openid offline` needed? And why there is a space in between? Do both of them needed to make it success?

Reply:
I tested myself, it seems needed... interesting...

Reply:
wait, hmm... seems it is not really needed `scope=offline` works as well with code.